include/kernel.h

```diff
@@ -137,6 +142,29 @@ enum k_objects {
 	K_OBJ_WORK,
 	K_OBJ_WORK_Q,
 
+	/* Driver subsystems */
```

Comment:
yes, there will be comprehensive documentation on how to set up new subsystems, kernel objects, and system calls.
writing this documentation is being punted until the implementation is mostly done since the design is still in flux in some areas (although we are a lot closer)